File: LeetCode/2537.py
class Solution:
    def countGood(self, nums: List[int], k: int) -> int:
        left, ans = 0, 0
        cnt_pair = 0
        cnt = Counter()
        for right, elem in enumerate(nums):
            cnt_pair += cnt[elem]
            cnt[elem] += 1
            while cnt_pair >= k:
                cnt_pair -= cnt[nums[left]] - 1
                cnt[nums[left]] -= 1
                left += 1
            ans += left
        return ans

File: LeetCode/2537.cpp
class Solution {
public:
	long long countGood(vector<int> &nums, int k) {
		long long ans = 0;
		int n = nums.size();
		std::unordered_map<int, int> cnt;
		long long curr_cnt = 0;
		for (int left = 0, right = 0; right < n; right++) {
			curr_cnt += cnt[nums[right]];
			++cnt[nums[right]];
			while (curr_cnt >= k) {
				curr_cnt -= cnt[nums[left]] - 1;
				--cnt[nums[left]];
				++left;
			}
			ans += left;
		}
		return ans;
	}
};
